hazelcast/src/test/java/com/hazelcast/nio/tcp/TcpIpConnectionManager_NewClientConnectTest.java

```diff
@@ -0,0 +1,142 @@
+package com.hazelcast.nio.tcp;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.ClientAuthenticationCodec;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.ConnectionType;
+import com.hazelcast.nio.Protocols;
+import com.hazelcast.test.AssertTask;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+import static com.hazelcast.client.ClientTypes.JAVA;
+import static com.hazelcast.nio.ConnectionType.JAVA_CLIENT;
+import static com.hazelcast.nio.tcp.TcpIpConnection_AbstractTest.getConnection;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+// Unfortunately the new client functionality relies on Node. So we can't test it in isolation.
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class TcpIpConnectionManager_NewClientConnectTest extends HazelcastTestSupport {
+
+    private static HazelcastInstance hz;
+    private static TcpIpConnectionManager connManagerA;
+    private static Address addressA;
+    private static Config config;
+    private Socket socket;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        setLoggingLog4j();
+        hz = Hazelcast.newHazelcastInstance();
+        connManagerA = (TcpIpConnectionManager) getConnectionManager(hz);
+        addressA = getNode(hz).getThisAddress();
+        config = getNode(hz).getConfig();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        Hazelcast.shutdownAll();
+    }
+
+    @After
+    public void tearDown() {
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException e) {
+            }
+        }
+    }
+
+    @Test
+    public void connectOldClient_whenJava() throws IOException {
```

Comment:
Ahh.. initially it was a single test for both old and new.. Since the old stuff is going to be removed eventually I decided to split up the test... so this is an artifact from that.